ui/webapp.py

```diff
@@ -5,11 +5,16 @@
 import pandas as pd
 from json import JSONDecodeError
 from pathlib import Path
-import streamlit as st
-from annotated_text import annotation
-from markdown import markdown
 
-from utils import haystack_is_ready, query, send_feedback, upload_doc, haystack_version, get_backlink
+try:
+    import streamlit as st
+    from annotated_text import annotation
+    from markdown import markdown
+
+    from ui.utils import haystack_is_ready, query, send_feedback, upload_doc, haystack_version, get_backlink
+except (ImportError, ModuleNotFoundError) as ie:
+    from haystack.utils.import_utils import _optional_component_not_installed
+    _optional_component_not_installed(__name__, "ui", ie)
```

Comment:
`_optional_component_not_installed` is used quite some time outside of `import_utils`. I'd recommend to make it look public or renaming it in general to `on_optional_component_not_installed`

Reply:
Hm that's true, but I still wouldn't like to expose it as public API for the end users :sweat_smile: In my opinion if the users are not supposed to use it, it should be marked private with `_`

Reply:
Yep, makes sense, you're totally right.